tests/main/upgrade-from-release/task.yaml

```
@@ -49,13 +49,14 @@ execute: |
 
     declare -A EXPECTED_SNAPD_VERSIONS=(
         ["24.04"]='2.65.3\+24.04'
-        ["22.04"]='2.55.3\+22.04'
-        ["20.04"]='2.44.3\+20.04'
+        ["22.04"]='2.65.3\+22.04'
+        ["20.04"]='2.65.3\+20.04'
         ["18.04"]='2.32.5\+18.04'
         ["16.04"]='2.48.3'
         ["14.04"]='2.37.4~14.04.1'
     )
-    apt list --installed snapd | MATCH "${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}"
+
+    apt list --installed snapd | MATCH "${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}\]"
```

Is this a typo?

---

No, check out the output from apt here:
```
ubuntu@noble:~$ apt list --installed snapd
Listing... Done
snapd/noble-security,now 2.63+24.04ubuntu0.1 amd64 [installed,upgradable to: 2.65.3+24.04]
```

We want to make sure we're checking on that last string, not the first one.

---

Ahh, nice catch. Perhaps a comment in the spread test saying as much would be useful.

Also I think there would be a problem here: what if the image already has the latest version installed, so it's
```
snapd/noble-security,now 2.65.3+24.04ubuntu0.1 amd64 [installed]
```

I think we need some sort of if-else or case check, something like:

```suggestion
    SNAPD_INSTALLED="$(apt list --installed snapd)"
    if echo "$SNAPD_INSTALLED" | MATCH 'upgradable to' ; then
        echo "$SNAPD_INSTALLED" | MATCH "${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}\]"
    else
        echo "$SNAPD_INSTALLED" | MATCH "${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}"
    fi
```

---

I think my suggestion may have been a bit naïve, you might need
```diff
- echo "$SNAPD_INSTALLED" | MATCH "upgradable to: ${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}\]$"
+ echo "$SNAPD_INSTALLED" | MATCH "upgradable to: .*${EXPECTED_SNAPD_VERSIONS[$VERSION_ID]}\]$"
```